core/src/main/scala/org/http4s/Query.scala

```diff
@@ -70,26 +84,15 @@ final class Query private (pairs: Vector[KeyValue])
     * If multiple values exist for a key, the first is returned. If
     * none exist, the empty `String` "" is returned.
     */
-  def params: Map[String, String] = new ParamsView(multiParams)
+  lazy val params: immutable.Map[String, String] =
+    CollectionCompat.mapValues(multiParams)(_.headOption.getOrElse(""))
```

Comment:
This is not the most efficient thing I can imagine, but `ParamsView` is not salvageable. Look at that `+`.